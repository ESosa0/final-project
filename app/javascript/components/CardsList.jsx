import React from 'react'
import SingleChoiceCard from './SingleChoiceCard'
import MultiChoiceCard from './MultiChoiceCard'
import IntroCard from './IntroCard'
import ScrollableAnchor, { goToAnchor } from '@ludois/react-scrollable-anchor'

const CardsList = ({ questions, currentStep, goToNextStep, addToSelections, onFinalStepSubmit }) => {
  console.log(questions)
  const questionElements = questions.map((question, index) => {
    const step = index+1
    if (step > currentStep) return false
    const CardItem = index === questions.length - 1
      ? MultiChoiceCard
      : SingleChoiceCard;

    const onNextClick = index === questions.length - 1
      ? onFinalStepSubmit
      : () => {
        goToNextStep(step+1)
        goToAnchor(questions[step].slug)
      }

    return ( 
      <ScrollableAnchor 
        id={question.slug}
        key={index}
        >
        <CardItem  
          {...question} 
          step={step}
          onNextClick={onNextClick}
          onChoiceSelection={addToSelections}
        />
      </ScrollableAnchor>
    )
  })
  
  return (
    <ul className="card-list">
      <IntroCard 
        onNextClick={() => {
          goToNextStep(1)
          goToAnchor(questions[0].slug)
        }}
      />
      {questionElements}
    </ul>
  )
}

export default CardsList
