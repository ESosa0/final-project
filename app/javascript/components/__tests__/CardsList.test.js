import CardsList from '../CardsList'
import {shallow, mount} from 'enzyme'
import toJSON from 'enzyme-to-json'
import React from 'react'

describe("CardsList Component", () => {
  it("renders a default cards list", () => {
    const questions = [
      {
        text: "What are you cooking?",
        choices: [
          {
            name: "fish",
            id: "1",
          }
        ],
        currentSelections: [],
        slug: "main-ingredient",
      },
      {
        text: "How are you cooking it?",
        choices: [
          {
            name: "fish",
            id: "1",
          }
        ],
        currentSelections: [],
        slug: "cooking-method", 
      },
    ]
    const component = shallow(<CardsList questions = {questions} />)
    expect(toJSON(component)).toMatchSnapshot()
  })
})
