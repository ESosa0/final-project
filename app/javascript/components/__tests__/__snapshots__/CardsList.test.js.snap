// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList Component renders a default cards list 1`] = `
<ul
  className="card-list"
>
  <IntroCard
    onNextClick={[Function]}
  />
  <ScrollableAnchor
    id="main-ingredient"
    key="0"
  >
    <SingleChoiceCard
      buttonText="Next"
      choices={
        Array [
          Object {
            "id": "1",
            "name": "fish",
          },
        ]
      }
      currentSelections={Array []}
      onNextClick={[Function]}
      slug="main-ingredient"
      step={1}
      text="What are you cooking?"
    />
  </ScrollableAnchor>
  <ScrollableAnchor
    id="cooking-method"
    key="1"
  >
    <MultiChoiceCard
      buttonText="Pair My Dish"
      choices={
        Array [
          Object {
            "id": "1",
            "name": "fish",
          },
        ]
      }
      currentSelections={Array []}
      slug="cooking-method"
      step={2}
      text="How are you cooking it?"
    />
  </ScrollableAnchor>
</ul>
`;
